pkg/compact/compact.go

```diff
@@ -1460,7 +1490,33 @@ func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 			go func() {
 				defer wg.Done()
 				for g := range groupChan {
-					shouldRerunGroup, _, err := g.Compact(workCtx, c.compactDir, c.planner, c.comp, c.blockDeletableChecker, c.compactionLifecycleCallback)
+					// Delete before merge: Get the metas for the current group
+					groupMetas := g.Metas()
+
+					// Delete before merge: Convert []*metadata.Meta to []metadata.Meta
+					plannedMetas := make([]metadata.Meta, len(groupMetas))
```

Comment:
Ok, I think we are just setting blocks of the group, instead of the planned blocks? We need planner.Plan output